pkg/arvo/lib/quiz.hoon

```diff
@@ -0,0 +1,61 @@
+|_  [eny=@uv runs=@ud]
+++  check
+  |=  vax=vase
+  =?  runs  =(0 runs)  100
+  =+  size=1
+  =+  sax=(slot 6 vax)
+  =+  run-i=0
+  =+  rng=~(. og eny)
+  |-  ^-  ?
+  ?:  =(run-i runs)
+      ~&  success-runs+run-i
+      %.y
+  =+  sam=(~(fill quiz [size rng]) sax)
+  =+  res=(slam vax sam)
+  ?:  =(+:res %.y)
+    :: Arbitrarily chosen growth pace.
+    =+  new-size=(add +(size) (div (mul size 2) 21))
+    $(run-i +(run-i), rng +:(rads:rng 1), size new-size)
+  ~&  fail-with-sam+q.sam
+  %.n
+++  quiz  |_  [size=@ud rng=_og]
+  ++  split-rng  ^+  [og og]
+    =+  bit-size=256
+    =^  bits-1  rng  (raws:rng bit-size)
+    [~(. og bits-1) ~(. og (raw:rng bit-size))]
+  ++  fill
+    |=  sax=vase
+    ^+  sax
+    =+  new-rng=+:(rads:rng 1)
+    ?+  p.sax  ~&(warn-unfill-sam+`type`p.sax sax)
+      %noun            sax(q (gen-noun))
+      [%atom p=* q=~]  sax(q (gen-atom p.p.sax))
+      [%atom *]        sax
+      [%cell p=* q=*]  =+  [rng-1 rng-2]=split-rng
+                       %=  sax
+                          q  :-  q:(fill(rng rng-2) (slot 2 sax))
+                                 q:(fill(rng rng-1) (slot 3 sax))
+                      ==
+      [%face p=* q=*]  sax(q q:(fill [p=q.p.sax q=q.sax]))
+    ==
+  ++  gen-noun
+    |.
```

Comment:
You don't need the extra core here.  Instead, the `|.` can close around `start-size`.
```hoon
  ++  gen-noun
    =+  start-size=size
    |.  ^-  noun
    ?:  (lte size 1)
      (rad:rng start-size)
    ...
```